update mgt-file-list with new attributes and options from file-upload. (#13861)
microsoftgraph/microsoft-graph-docs: concepts/toolkit/components/file-list.md

| page-size | pageSize | A number value to indicate the maximum number of files to render on each page. |
| file-extensions | fileExtensions | An array of file extensions used to filter files to show. |
| hide-more-files-button | hideMoreFilesButton | A boolean to indicate whether to show a button to render more files. |

> | enable-file-upload | enableFileUpload | Boolean to enable or disable file upload extension, default is false.  |

Reviewer: ```suggestion
| enable-file-upload | enableFileUpload | Boolean to enable or disable file upload functionality. The default value is `false`.  |
```

> | excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also provide `enable-file-upload` attribute. |

Reviewer: ```suggestion
| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also set the `enable-file-upload` attribute to `true`. |
```

> | excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. |
> | max-file-size | maxFileSize | Number to restrict upload size (KB).  |
> | max-upload-file | maxUploadFile | Number, default value are 10 files. |

Reviewer 2: ```suggestion
| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also provide `enable-file-upload` attribute. |
| max-file-size | maxFileSize | Number to restrict upload size (KB). Must also provide `enable-file-upload` attribute. |
| max-upload-file | maxUploadFile | Number, default value are 10 files. Must also provide `enable-file-upload` attribute. |
```

> | max-file-size | maxFileSize | Number to restrict upload size (KB). Must also provide `enable-file-upload` attribute. |

Reviewer: ```suggestion
| max-file-size | maxFileSize | A number representing the maximum file upload size (KB). Must also set the `enable-file-upload` attribute to `true`. |
```

> | max-upload-file | maxUploadFile | Number, default value are 10 files. Must also provide `enable-file-upload` attribute. |

Reviewer: ```suggestion
| max-upload-file | maxUploadFile | A number representing the maximum number of files allowed to be uploaded. The default value is `10` files. Must also set the `enable-file-upload` attribute to `true`. |
```

<mgt-file-list insight-type="shared"></mgt-file-list>
```

> The following example enable the file upload feature.

Reviewer 2: ```suggestion
The following example enables the file upload feature.
```

```html
<mgt-file-list enable-file-upload></mgt-file-list>
```

> The following example limit upload to 5 files.

Reviewer 2: ```suggestion
The following example limits maximum number of files that can be uploaded to 5.
```

> The following example limits maximum number of files that can be uploaded to 5.

Reviewer: ```suggestion
The following example limits the maximum number of files that can be uploaded to 5.
```

```html
<mgt-file-list max-upload-file="5" enable-file-upload></mgt-file-list>
```

> The following example limit upload file size to 10000 KB.

Reviewer 2: ```suggestion
The following example limits maximum file size that can be uploaded to 10000 KB.
```

> The following example limits maximum file size that can be uploaded to 10000 KB.

Reviewer: ```suggestion
The following example limits the maximum file size that can be uploaded to 10000 KB.
```

```html
<mgt-file-list max-file-size="10000" enable-file-upload></mgt-file-list>
```

> The following example exclude upload of files with file extensions ".doc,.pdf".

Reviewer 2: ```suggestion
The following example excludes upload of files with file extensions ".doc,.pdf".
```

| Configuration | Permissions | API |
| ------------- | ----------------- | --- |

> | Default (no identifiers or query provided) | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /me/drive/root/children` <br /> `PUT /me/drive/root:/{filename}:/content` <br /> `POST /me/drive/root:/{filename}:/createUploadSession` |

Reviewer 2: @aaclage I think these need to be separated to their own row. They depend on the attribute `{enable-file-upload}`.
```suggestion
| Default (no identifiers or query provided) | Files.Read, Files.Read.All, Sites.Read.All | `GET /me/drive/root/children` |
| Provide `{enable-file-upload}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /me/drive/root/children` <br /> `PUT /me/drive/root:/{filename}:/content` <br /> `POST /me/drive/root:/{filename}:/createUploadSession` |
```

> | Provide `{drive-id}` AND `{item-id}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /drives/{drive-id}/items/{item-id}/children` <br /> `PUT /drives/{drive-id}/items/{item-id}:/{filename}:/content` <br /> `POST /drives/{drive-id}/items/{item-id}:/{filename}:/createUploadSession` |

Reviewer 2: Similar here and below:

```suggestion
| Provide `{drive-id}` AND `{item-id}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /drives/{drive-id}/items/{item-id}/children`|
| Provide `{drive-id}` AND `{item-id}` AND `{enable-file-upload}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /drives/{drive-id}/items/{item-id}/children` <br /> `PUT /drives/{drive-id}/items/{item-id}:/{filename}:/content` <br /> `POST /drives/{drive-id}/items/{item-id}:/{filename}:/createUploadSession` |
```

Reviewer 2: Need to change the sections following this part too. :)
